File: weagle-api/src/main/java/com/weagle/entity/Idea.java
package com.weagle.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Document(collection = "ideas")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Idea {

    @Id
    private String id;

    private String title;

    private String description;

    private String createdBy;

    private boolean approved;

    private boolean highPriority;

    private LocalDateTime createdAt;
}

File: app/src/main/java/com/fiapos/weagle/domain/models/Idea.kt
package com.fiapos.weagle.domain.models

import java.time.LocalDate

data class Idea(
    val id: String = "",
    val title: String,
    val description: String,
    val type: IdeaType,
    val status: IdeaStatus = IdeaStatus.PENDING,
    val createdBy: String,
    val createdAt: LocalDate?,
    val isEdited: Boolean = false,
    val votes: Int = 0,
    val isSelectable: Boolean = false,
    val isSelected: Boolean = false
)